- en: 原文：[https://dev.mysql.com/doc/refman/8.0/en/commands-out-of-sync.html](https://dev.mysql.com/doc/refman/8.0/en/commands-out-of-sync.html)
  id: totrans-0
  prefs:
  - PREF_BQ
  type: TYPE_NORMAL
  zh: 原文：[https://dev.mysql.com/doc/refman/8.0/en/commands-out-of-sync.html](https://dev.mysql.com/doc/refman/8.0/en/commands-out-of-sync.html)
- en: B.3.2.12 Commands out of sync
  id: totrans-1
  prefs:
  - PREF_H4
  type: TYPE_NORMAL
  zh: B.3.2.12 命令不同步
- en: If you get `Commands out of sync; you can't run this command now` in your client
    code, you are calling client functions in the wrong order.
  id: totrans-2
  prefs: []
  type: TYPE_NORMAL
  zh: 如果在客户端代码中出现`Commands out of sync; you can't run this command now`，说明你在错误的顺序中调用客户端函数。
- en: This can happen, for example, if you are using [`mysql_use_result()`](/doc/c-api/8.0/en/mysql-use-result.html)
    and try to execute a new query before you have called [`mysql_free_result()`](/doc/c-api/8.0/en/mysql-free-result.html).
    It can also happen if you try to execute two queries that return data without
    calling [`mysql_use_result()`](/doc/c-api/8.0/en/mysql-use-result.html) or [`mysql_store_result()`](/doc/c-api/8.0/en/mysql-store-result.html)
    in between.
  id: totrans-3
  prefs: []
  type: TYPE_NORMAL
  zh: 这可能发生在你使用[`mysql_use_result()`](/doc/c-api/8.0/en/mysql-use-result.html)并在调用[`mysql_free_result()`](/doc/c-api/8.0/en/mysql-free-result.html)之前尝试执行新查询时。也可能发生在你尝试执行两个返回数据的查询而没有在中间调用[`mysql_use_result()`](/doc/c-api/8.0/en/mysql-use-result.html)或[`mysql_store_result()`](/doc/c-api/8.0/en/mysql-store-result.html)。
